File: Event/src/main/java/tn/esprit/spring/event/model/Event.java
package tn.esprit.spring.event.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idEvent;

    private String title;
    private String description;
    private String location;

    private LocalDate dateEvent;

    private float goalAmount;
    private float collectedAmount;

    @Enumerated(EnumType.STRING)
    private EventType type;
}

File: charity/src/app/models/event.ts
export type EventType = 'DONATION' | 'EDUCATION' | 'ENVIRONMENT' | 'HEALTH' | 'OTHER';

export interface Event {
  idEvent?: number;
  title: string;
  description: string;
  location: string;
  dateEvent: string; // YYYY-MM-DD
  goalAmount: number;
  collectedAmount: number;
  type: EventType;
}
